pkg/nemesis/network_partition.go

```diff
@@ -0,0 +1,160 @@
+	return g.name
+}
+
+func partitionNodes(nodes []cluster.Node, n int) []*core.NemesisOperation {
+	ops := make([]*core.NemesisOperation, len(nodes))
+
+	// randomly shuffle the indices and get the first n nodes to be partitioned.
+	indices := shuffleIndices(len(nodes))
+
+	var onePartNodes []cluster.Node
+	var anotherPartNodes []cluster.Node
+
+	for i := 0; i < n; i++ {
+		onePartNodes = append(onePartNodes, nodes[indices[i]])
+	}
+	for i := n; i < len(nodes); i++ {
+		anotherPartNodes = append(anotherPartNodes, nodes[indices[i]])
+	}
+
+	ops[0] = &core.NemesisOperation{
+		Type:        core.NetworkPartition,
+		InvokeArgs:  []interface{}{onePartNodes, anotherPartNodes},
+		RecoverArgs: []interface{}{onePartNodes, anotherPartNodes},
+		RunTime:     time.Second * time.Duration(rand.Intn(120)+60),
+	}
+
+	return ops
+}
+
+// NewNetworkPartitionGenerator creates a generator.
+// Name is partition-one, etc.
+func NewNetworkPartitionGenerator(name string) core.NemesisGenerator {
+	return networkPartitionGenerator{name: name}
+}
+
+type networkPartition struct {
+	k8sNemesisClient
+}
+
+func (n networkPartition) Invoke(ctx context.Context, _ cluster.Node, args ...interface{}) error {
+	onePart, anotherPart := extractTwoParts(args...)
+	return n.cli.ApplyNetChaos(&chaosv1alpha1.NetworkChaos{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", onePart[0].Namespace, chaosv1alpha1.PartitionAction),
+			Namespace: onePart[0].Namespace,
+		},
+		Spec: chaosv1alpha1.NetworkChaosSpec{
+			Action: chaosv1alpha1.PartitionAction,
+			Mode:   chaosv1alpha1.AllPodMode,
+			Selector: chaosv1alpha1.SelectorSpec{
+				Pods: map[string][]string{
+					onePart[0].Namespace: extractPodNames(onePart),
+				},
+			},
+			Direction: chaosv1alpha1.Both,
+			Target: chaosv1alpha1.PartitionTarget{
+				TargetSelector: chaosv1alpha1.SelectorSpec{
+					Pods: map[string][]string{
+						anotherPart[0].Namespace: extractPodNames(anotherPart),
+					},
+				},
+				TargetMode: chaosv1alpha1.AllPodMode,
+			},
+		},
+	})
+}
+
+func (n networkPartition) Recover(ctx context.Context, _ cluster.Node, args ...interface{}) error {
+	onePart, anotherPart := extractTwoParts(args...)
+	return n.cli.ApplyNetChaos(&chaosv1alpha1.NetworkChaos{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-%s", onePart[0].Namespace, chaosv1alpha1.PartitionAction),
+			Namespace: onePart[0].Namespace,
+		},
+		Spec: chaosv1alpha1.NetworkChaosSpec{
+			Action: chaosv1alpha1.PartitionAction,
+			Mode:   chaosv1alpha1.AllPodMode,
+			Selector: chaosv1alpha1.SelectorSpec{
+				Pods: map[string][]string{
+					onePart[0].Namespace: extractPodNames(onePart),
+				},
+			},
+			Direction: chaosv1alpha1.Both,
+			Target: chaosv1alpha1.PartitionTarget{
+				TargetSelector: chaosv1alpha1.SelectorSpec{
+					Pods: map[string][]string{
+						anotherPart[0].Namespace: extractPodNames(anotherPart),
+					},
+				},
+				TargetMode: chaosv1alpha1.AllPodMode,
+			},
+		},
+	})
+}
+
```

Comment:
The `NetworkChaos` templates is almost the same...why not just create a function like `podTag`

Reply:
IMO, it's early to extract such function